src/lib.rs

```diff
@@ -303,31 +344,6 @@ impl<T> ThinVec<T> {
     }
     */
 
-    fn reserve_one_more(&mut self) {
-        // TODO: capacity overflow logic?
-        let old_cap = self.capacity();
-        let old_len = self.len();
-
-        if old_cap <= old_len {
-            let elem_size = mem::size_of::<T>();
-
-            // "Infinite" capacity for ZSTs
-            let new_cap = if elem_size == 0 { !0 } else 
-                          if old_cap == 0 { 4 } else 
```

Comment:
Fixed.  The "minimum 4" optimization is now applied in `reserve`.  The "infinite size for ZSTs" optimization is now applied in `header_with_capacity`.